Common/Utils/include/CommonUtils/BoostHistogramUtils.h

```diff
@@ -0,0 +1,255 @@
+  static Double_t kTol = mat.GetTol();
+  fitter.StoreData(kFALSE);
+  fitter.ClearPoints();
+  TVectorD par(3);
+  TVectorD sigma(3);
+  TMatrixD A(3, 3);
+  TMatrixD b(3, 1);
+  T rms = TMath::RMS(nBins, arr);
+  T max = TMath::MaxElement(nBins, arr);
+  T binWidth = (xMax - xMin) / T(nBins);
+
+  Float_t meanCOG = 0;
+  Float_t rms2COG = 0;
+  Float_t sumCOG = 0;
+
+  Float_t entries = 0;
+  Int_t nfilled = 0;
+
+  param.resize(4);
+  param[0] = 0.;
+  param[1] = 0.;
+  param[2] = 0.;
+  param[3] = 0.;
+
+  for (Int_t i = 0; i < nBins; i++) {
+    entries += arr[i];
+    if (arr[i] > 0) {
+      nfilled++;
+    }
+  }
+
+  // TODO: Check why this is needed
+  if (max < 4) {
+    return -4;
+  }
+  if (entries < 12) {
+    return -4;
+  }
+
+  if (rms < kTol) {
+    return -4;
+  }
+
+  param[3] = entries;
+
+  for (auto iter = first, axisiter = axisfirst; iter != last; iter++, axisiter++) {
+  }
+
+  auto nbins = last - first;
+  const double binWidth = double(xMax - xMin) / double(nBins);
+  int ibin = 0;
+  Int_t npoints = 0;
+  for (auto iter = first, axisiter = axisfirst; iter != last; iter++, axisiter++) {
+    if (nbins > 1) {
+      const double x = (*axisiter + *(axisiter + 1)) / 2.;
+      const double y = *iter;
+      const double ey = std::sqrt(y);
+      fitter.AddPoint(x, y, ey);
+      if (npoints < 3) {
+        A(npoints, 0) = 1;
+        A(npoints, 1) = x;
+        A(npoints, 2) = x* xor_eq ;
+        b(npoints, 0) = y;
+        meanCOG += x * nbins;
+        rms2COG += x * nbins * x;
+        sumCOG += nbins;
+      }
+      npoints++;
+    }
+  }
+
+  Double_t chi2 = 0;
+  if (npoints >= 3) {
+    if (npoints == 3) {
+      //analytic calculation of the parameters for three points
+      A.Invert();
+      TMatrixD res(1, 3);
+      res.Mult(A, b);
+      par[0] = res(0, 0);
+      par[1] = res(0, 1);
+      par[2] = res(0, 2);
+      chi2 = -3.;
+    } else {
+      // use fitter for more than three points
+      fitter.Eval();
+      fitter.GetParameters(par);
+      fitter.GetCovarianceMatrix(mat);
+      chi2 = fitter.GetChisquare() / Double_t(npoints);
+    }
+    if (TMath::Abs(par[1]) < kTol) {
+      return -4;
+    }
+    if (TMath::Abs(par[2]) < kTol) {
+      return -4;
+    }
+
+    param[1] = T(par[1] / (-2. * par[2]));
+    param[2] = T(1. / TMath::Sqrt(TMath::Abs(-2. * par[2])));
+    Double_t lnparam0 = par[0] + par[1] * param[1] + par[2] * param[1] * param[1];
+    if (lnparam0 > 307) {
+      return -4;
+    }
+    param[0] = TMath::Exp(lnparam0);
+
+    return chi2;
+  }
+
+  if (npoints == 2) {
+    //use center of gravity for 2 points
+    meanCOG /= sumCOG;
+    rms2COG /= sumCOG;
+    param[0] = max;
+    param[1] = meanCOG;
+    param[2] = TMath::Sqrt(TMath::Abs(meanCOG * meanCOG - rms2COG));
+    chi2 = -2.;
+  }
+  if (npoints == 1) {
+    meanCOG /= sumCOG;
+    param[0] = max;
+    param[1] = meanCOG;
+    param[2] = binWidth / TMath::Sqrt(12);
+    chi2 = -1.;
+  }
+  return chi2;
+}
+
```

Comment:
Maybe having a dedicated function `fitGaus(boost::histogram ...)` as frontend which uses the functions above as backend could be helpful, if that is possible. Might need to be templated with the boost histogram as template argument.